server/src/main/java/org/elasticsearch/cluster/metadata/MetaDataIndexStateService.java

```diff
@@ -494,4 +536,16 @@ private static int getTotalShardCount(ClusterState state, Index index) {
         return indexMetaData.getNumberOfShards() * (1 + indexMetaData.getNumberOfReplicas());
     }
 
+    /**
+     * @return Generates a {@link ClusterBlock} that blocks read and write operations on soon-to-be-closed indices. The
+     * cluster block is generated with the id value equals to {@link #INDEX_CLOSED_BLOCK_ID} and a unique UUID.
+     */
+    public static ClusterBlock createIndexClosedBlock() {
+        final String description = "Index is blocked due to on-going index closing operation. Note that the closing process can take " +
+            "time and writes operations are blocked in the meantime. Execute an open index request to unblock the index and allow writes " +
+            "operation again. Execute a new close index request will try to close the index again.";
+        return new ClusterBlock(INDEX_CLOSED_BLOCK_ID, UUIDs.randomBase64UUID(), description , false, false, false,
+            RestStatus.FORBIDDEN, ClusterBlockLevel.READ_WRITE);
```

Comment:
should we only block writes here instead of both reads and writes? The error message will need updating if we are to also block reads...

Reply:
Yes, we should only block writes.